File: Java/DecoratorPattern/src/coffee/Latte.java
package coffee;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 27-05-2021
 * Time: 04:25 AM
 */
public class Latte extends Coffee{

    public Latte(){
        description = "Latte : Rs "+cost();
    }

    @Override
    public double cost() {
        return 21.5;
    }
}

File: Kotlin/DecoratorPattern/src/coffee/Latte.kt
package coffee

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 27-05-2021
 *   Time: 04:55 AM
 *
 */
class Latte : Coffee() {

    init {
        description = "Latte : Rs ${cost()}"
    }

    override fun cost(): Double {
        return 20.0
    }

}